src/main.rs

```diff
@@ -262,16 +264,33 @@ mod app {
         telemetry_task::spawn_after(((telemetry_period * 1000.0) as u64).millis()).unwrap();
     }
 
-    #[task(priority = 1, shared=[network, settings])]
+    #[task(priority = 1, shared=[network, settings, ch_temperature])]
     fn mqtt_interlock(mut c: mqtt_interlock::Context) {
         let interlock = c
             .shared
             .settings
             .lock(|settings| settings.interlock.clone());
         if interlock.armed {
+            let tempemperatures = c.shared.ch_temperature.lock(|temp| *temp);
+            let mut tripped = false;
+            for (i, ((temp, lower), upper)) in tempemperatures
+                .iter()
+                .map(|temp| *temp as f32)
+                .zip(interlock.temperature_limit_lower)
+                .zip(interlock.temperature_limit_upper)
+                .enumerate()
+            {
+                if temp < lower || temp > upper {
+                    tripped = true;
+                    defmt::error!(
+                        "channel {:?} temperature out of range, interlock tripped!",
+                        i
+                    );
+                }
+            }
```

Comment:
```suggestion
            let tripped = temperatures.iter().zip(interlock.temperature_limits).enumerate().any(|(i, (&temp, &limits))| {
                let t = !(limits[0]..limits[1]).contains(temp);
                if t {
                    defmt::error!(
                        "channel {:?} temperature out of range, interlock tripped!",
                        i
                    );
                }
                t
            })
```

That's a bit more idiomatic. And once you decide you can remove the debugging prints, then this becomes a very short one-liner.

Reply:
Yes, I just thought this debugging information might be valuable at some point.